android/BOINC/app/src/main/java/edu/berkeley/boinc/rpcExtern/RpcSettingsData.kt

```diff
@@ -0,0 +1,188 @@
+package edu.berkeley.boinc.rpcExtern
+
+import android.os.Parcel
+import android.os.Parcelable
+import kotlinx.android.parcel.Parcelize
+
+class RpcSettingsDatax {
+    val DEFAULT_PORT = 31416
+    val data = RpcSettingsData()
+    var validData = false
+    var ipAllowedList = ArrayList<String>()
+
+    fun set(enabled : Boolean, encryption : Boolean, passwrd : String, port :String, ip1 : String, ip2 : String, ip3 : String , ip4 : String)
+    {
+        validData = true
+        data.externEnabled = enabled
+        data.externEncryption = encryption
+        data.externPasswrd = passwrd
+
+        try {
+            if (port.isNotEmpty()) {
+                data.externPort = port.toInt()
+            }
+            else
+            {
+                data.externPort = DEFAULT_PORT
+            }
```

Comment:
```suggestion
            data.externPort = if (port.isNotEmpty()) port.toInt() else DEFAULT_PORT
```

Reply:
I rather not, I do a lot of testing with the debugger and my way is easy the other is hard to debug.
Not much shorter, not easier to read, these are all personal things, not effecting the code.